JoinColony/colonyDapp
# Connect Nft Picker to Existing safes

## Description

This PR connects the Nft Picker to the existing Safes in the colony. 

You can add the following safes with NFTS to your colony in order to test (both Rinkeby):

1. [0x05b990D01A2f86852feD1E7d609dd3E52aA9EA27](https://gnosis-safe.io/app/rin:0x05b990D01A2f86852feD1E7d609dd3E52aA9EA27)
1. [0x358E6D7C8Fa5e55fc6A0D50E23b875c6c49fF09A](https://gnosis-safe.io/app/rin:0x358E6D7C8Fa5e55fc6A0D50E23b875c6c49fF09A)

And this one doesn't have NFTs (also Rinkeby):
1. 0x0DA0C3e52C977Ed3cBc641fF02DD271c3ED55aFe

To test:
1. Add a safe with and without nfts
3. Go to safe control
4. Select a safe and select "Transfer NFT"
5. Should display loading state while we fetch nft data. If there's none, should display relevant message. If there's an error fetching (e.g. no internet connection), should also display relevant message. 
6. Once data is fetched, you should be able to select an nft in the safe. If it has an image uri, this should show in the avatar component, and the contract name and id should be correctly inputted. 

It also does a little bit of housekeeping:
1. Removing "testnet" from Goerli and Rinkeby. I added these in an earlier PR thinking they'd be useful but they just take up unnecessary space in the safe picker input and it's also not how Gnosis Safe refers to them. So reverting.
7. Updates validation schema for `safe`
8. renamed `getSafeCoreSDK.ts` to `safeHelpers.ts` and added a few utility functions.

**Changes** 🏗

* Wires up the NFT picker to the safes that exist in the colony.

## Screenshot

![nfta](https://user-images.githubusercontent.com/64402732/189651478-109226e3-66c7-49f1-b7e7-529bcdf052ff.png)

Resolves #3835
Contributes to #3722
